src/main/kotlin/org/rust/ide/intentions/ConvertFunctionToClosureIntention.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.parentOfType
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.resolve.ref.MethodResolveVariant
+
+class ConvertFunctionToClosureIntention : RsElementBaseIntentionAction<ConvertFunctionToClosureIntention.Context>() {
+
+    override fun getText(): String = "Convert function to closure"
+
+    data class Context(
+        val targetFunction: RsFunction,
+    )
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): Context? {
+        // We try to find a function declaration
+        val possibleTarget = element.ancestorStrict<RsFunction>() ?: return null
```

Comment:
I suggest to only make the intention available in the function signature (e.g. its identifier and parameter list). See https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/ide/refactoring/changeSignature/RsChangeSignatureHandler.kt#L28 for inspiration (basically the same code could work, it you remove the cases for `RsCallExpr` and `RsMethodCall`.

Reply:
I've changed this for both intentions as well - thank you for the hints on how to test that! For the closure -> function intention, I had to hack a little bit to make it feel like I thought it should. Please let me know if there's a better way of excluding the part after `}`.